File: nodeServer/java/GetCategories.java
import java.io.*;
import java.net.*;

import arenaapi.*;
import json.*;

public class GetCategories {

    public static void main(String[] arguments) throws IOException, URISyntaxException {

        // Parameters could be hard coded, from command line, etc
        String url = System.getenv("ARENA_API_URL");
        String email = System.getenv("ARENA_API_EMAIL");
        String password = System.getenv("ARENA_API_PASSWORD");
        String workspaceID = System.getenv("ARENA_API_WORKSPACEID");
        String log = System.getenv("ARENA_API_LOG");

        ArenaAPI arenaAPI = new ArenaAPI();
        arenaAPI.url = url;
        arenaAPI.log = log.length() > 0 && "1TtYy".indexOf(log.substring(0, 1)) != -1;
        boolean loggedIn = false;
        try {
            JSONObject args = new JSONObject();
            args.put("email", email);
            args.put("password", password);
            if (workspaceID.length() > 0)
                args.put("workspaceId", workspaceID);
            arenaAPI.apiCall("login", null, args);
            loggedIn = true;
            JSONObject result = (JSONObject) arenaAPI.apiCall("getItemCategories", null, null);
            System.out.println(result.toStringIndent(""));
        } catch (ArenaAPIException e) {
            arenaAPI.showResult(e.getMessage(), e.statusCode, e.errors, null);
        } finally {
            if (loggedIn) {
                try {
                    arenaAPI.apiCall("logout", null, null);
                } catch (ArenaAPIException e) {
                }
            }
        }
    }
}

File: nodeServer/node/getcategories.js
var arenaapi = require('./arenaapi.js');

// Parameters could be hard coded, from command line, etc
var url = process.env.ARENA_API_URL;
var email = process.env.ARENA_API_EMAIL;
var password = process.env.ARENA_API_PASSWORD;
var workspaceID = process.env.ARENA_API_WORKSPACEID;
var log = process.env.ARENA_API_LOG;

arenaapi.url = url;
arenaapi.log = '1TtYy'.indexOf(log[0]) != -1;

var args = workspaceID ? {email: email, password: password, workspaceId: workspaceID} : {email: email, password: password};

arenaapi.login({}, args, function(statusCode, errors, result) {
  if (errors == null) {
    arenaapi.getItemCategories({}, function(statusCode, errors, result) {
      console.log(errors || result);
      arenaapi.logout({}, function(statusCode, errors, result) {
      });
    });
  } else
    console.log(errors);
});
